.github/workflows/build-connector-template.yml

```diff
@@ -0,0 +1,70 @@
+name: Build (Connector)
+
+on:
+  workflow_call:
+    inputs:
+      repo-name:
+        required: true
+        type: string
+      additional-build-flags:
+        required: false
+        type: string
+        default: ""
+      additional-test-flags:
+        required: false
+        type: string
+        default: ""
+
+
+jobs:
+    ubuntu-build:
+        name: Build
+        runs-on: ubuntu-latest
+        concurrency:
+          group: ${{ github.head_ref }}-ubuntu-build
+          cancel-in-progress: true
+        steps:
+            -   name: Checkout Repository
+                uses: actions/checkout@v3
+
+            - name: Set Up Ballerina
+              uses: ballerina-platform/setup-ballerina@v1.1.0
+              with:
+                version: latest
+
+            -   name: Set up JDK 17
+                uses: actions/setup-java@v3
+                with:
+                    distribution: 'temurin'
+                    java-version: 17.0.7
+            
+            -   name: Set ENV Variables
+                run: |
+                  echo -e '${{ toJson(secrets) }}' | jq -r 'to_entries[] | .key + "=" + .value' >> $GITHUB_ENV 
+
+            -   name: Build the Package
+                env:
+                    packageUser: ${{ github.actor }}
+                    packagePAT: ${{ secrets.GITHUB_TOKEN }}
+                run: |
+                  ./gradlew build -x test ${{ inputs.additional-build-flags }}
+                  ./gradlew test ${{ inputs.additional-test-flags }}
+
+            -   name: Generate Codecov Report
+                uses: codecov/codecov-action@v3
+
+            # Send notification when build fails
+            - name: Alert notifier on failure
+              if: failure() && (github.event.action == 'check_connector_for_breaking_changes')
+              run: |
+                curl -X POST \
+                'https://api.github.com/repos/ballerina-platform/ballerina-release/dispatches' \
+                --header 'Accept: application/vnd.github.v3+json' \
+                --header 'Authorization: Bearer ${{ secrets.BALLERINA_BOT_TOKEN }}' \
+                --data-raw '{
+                  "event_type": "notify-ballerinax-connector-build-failure",
+                  "client_payload": {
+                    "repoName": "${{ inputs.repo-name }}",
+                    "workflow": "CI"
+                  }
+                }'
```

Comment:
We might need to use a centralized script to build all the packages and send a single notification on failure, to avoid noise. But I think we can keep this for now, until we come up with a central script, WDYT @NipunaRanasinghe?